File: src/Chess_common/HistoryItem.java
package Chess_common;

import Chess_pieces.Pawn;
import Chess_pieces.Piece;

public class HistoryItem {
    private Field from;
    private Field to;
    private Piece target_to;
    private Piece exchange;

    public HistoryItem(Field from, Field to, Piece target_to) {
        this.from = from;
        this.to = to;
        this.target_to = target_to;
        this.exchange = null;
    }

    public Field getFrom() {
        return from;
    }

    public Field getTo() {
        return to;
    }

    public Piece getTarget_to() {
        return target_to;
    }

    public void setExchange(Piece piece){
        this.exchange = piece;
    }

    public Piece getExchange() {
        return exchange;
    }
}

File: src/Chess_common/HistoryItem.cpp
/*!
 * @authors Michal Janů (xjanum03), Richard Gajda (xgajda06)
 */

#include "HistoryItem.h"

using namespace std;

HistoryItem::HistoryItem() = default;

void HistoryItem::addValues(Field *from, Field *to, Piece *target_to){
    this->from = from;
    this->to = to;
    this->target_to = target_to;
    this->exchange = nullptr;
}

Field* HistoryItem::getFrom() {
    return this->from;
}

Field* HistoryItem::getTo() {
    return this->to;
}

Piece* HistoryItem::getTarget_to() {
    return this->target_to;
}

void HistoryItem::setExchange(Piece* piece){
    this->exchange = piece;
}

Piece* HistoryItem::getExchange() {
    return this->exchange;
}

